packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -2938,3 +2926,128 @@ class _RenderCupertinoDialogActions extends RenderBox
     return defaultHitTestChildren(result, position: position);
   }
 }
+
+// A column layout with two widgets, where the top widget expands vertically as
+// needed, and the bottom widget has a minimum height.
+//
+// Both child widgets stretch horizontally to the parent's maximum width
+// constraint, with vertical space allocated in this priority:
+//
+//  1. The `bottom` widget receives its requested height, up to a
```

Comment:
I can't parse this sentence.  according to the code height <= `bottomMinHeight`.
should it be call  bottomMaxHeight

Reply:
Yes it should be `bottomMaxHeight`. My bad!

Reply:
It might be easier to understand it from the other way.
* If both widgets fit in the container, great, render both in their intrinsic height.
* If not, then prioritize reducing the bottom widget's height.
  * However, the bottom widget can not be reduced too much. It has a `minHeight`. If the bottom widget has been reduced to `minHeight`, then further reduction has to come from the top widget.